### DISCUSSION TITLE
What can I do with python?

### DISCUSSION THREAD

**Discussion by Alhonainy** [apache/iotdb#4076] [Q&A]
I am new to this field however I really like to use IoTDB for my projects,
so, I can create IoTDB, create storage groups, timeseries … etc, however I found many features supported by Java, so
my question is what else can I do with python? 
For example: do you have tools to analysis data by integrating with Hadoop and Spark. 
How to management solution by integrating IoTDB with Grafana?

Thank you in advance.

**Comment by jixuan1989** [comment]
Hi,

1. Using Python for analysis: the python API has supported Pandas.
2. For hadoop and Spark, current it only support Java.. (I did not tried pySpark..)  @JulianFeinauer  @sunjincheng121  maybe can provide more info.
3. Yes it is easy to integrate IoTDB with Grafana, see: https://iotdb.apache.org/UserGuide/Master/Ecosystem%20Integration/Grafana.html

**Reply by HTHou** [reply]
> For hadoop and Spark, current it only support Java.. (I did not tried pySpark..) @JulianFeinauer @sunjincheng121 maybe can provide more info.

Hi, I see there is another discussion about connecting PySpark with IoTDB and it finally worked.
- https://github.com/apache/iotdb/discussions/3851
